src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/LdapAccessPolicyDAO.scala

```diff
@@ -108,11 +113,16 @@ class LdapAccessPolicyDAO(
 
   override def deletePolicy(policy: FullyQualifiedPolicyId): IO[Unit] = executeLdap(IO(ldapConnectionPool.delete(policyDn(policy))))
 
-  override def loadPolicy(resourceAndPolicyName: FullyQualifiedPolicyId): IO[Option[AccessPolicy]] = executeLdap(
-    IO(Option(ldapConnectionPool.getEntry(policyDn(resourceAndPolicyName))).map(unmarshalAccessPolicy))
-  )
+  override def loadPolicy(resourceAndPolicyName: FullyQualifiedPolicyId): IO[Option[AccessPolicy]] = {
+    for {
+      entry <- executeLdap(IO(ldapConnectionPool.getEntry(policyDn(resourceAndPolicyName))))
+      policy <- Option(entry).traverse(unmarshalAccessPolicy)
+    } yield {
+      policy
+    }
+  }
 
-  private def unmarshalAccessPolicy(entry: Entry): AccessPolicy = {
+  private def unmarshalAccessPolicy(entry: Entry): IO[AccessPolicy] = IO {
```

Comment:
I'd still prefer not to promote the method all the way to `IO`, but instead using `Either` since this is really just pure computation.

For consistency, I think all `unmarshalxxx` should return `Either`, at call site, we can turn it into `IO` with `IO.fromEither`